test/test_interpreter.cpp

```diff
@@ -645,4 +645,33 @@ TEST_SUITE("xutils_handler"){
         }
     }
 }
-#endif
\ No newline at end of file
+#endif
+
+TEST_SUITE("complete_request")
+{
+    TEST_CASE("completion_test")
+    {
+        std::vector<const char*> Args = {/*"-v", "resource-dir", "....."*/};
+        xcpp::interpreter interpreter((int)Args.size(), Args.data());
+
+        nl::json execute = interpreter.execute_request("#include <iostream>", false, false, nl::json::object(), false);
+
+        REQUIRE(execute["status"] == "ok");
+
+        std::string code = "st";
+        int cursor_pos = 2;
+        nl::json result = interpreter.complete_request(code, cursor_pos);
+
+        REQUIRE(result["cursor_start"] == 0);
+        REQUIRE(result["cursor_end"] == 2);
+        REQUIRE(result["status"] == "ok");
+        bool found = false;
+        for(auto& r : result["matches"]) {
+            if(r == "static" || r == "struct") {
+                found = true;
+                break;
+            }
+        }
+        REQUIRE(found == true);
```

Comment:
As written, everything will be OK if the function returns at least one of the expected results. I'm not sure this is a good approach for this test.
It is easiest to check if the field converted to a string is equal to {"struct", "strict"}, but that way there is a danger that in a newer version of the library (iostream) we will have more possible values, for example "stone". Then of course we will have to adjust the test.
Alternatively, we can check that the function returns everything expected without requiring that there are no other values, like so:
```suggestion
        size_t found = 0;
        for (auto& r : result["matches"]) {
            if (r == "static" || r == "struct") {
                found++;
            }
        }
        REQUIRE(found == 2);
```

Reply:
Makes sense.